src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -1374,6 +1374,7 @@ const static EnumMap<GameCommand> ActionNameToType = {
     { "ridesetvehicle", GameCommand::SetRideVehicles },
     { "scenariosetsetting", GameCommand::EditScenarioOptions },
     { "cheatset", GameCommand::Cheat },
+    { "scenerysetrestricted", GameCommand::SetRestrictedScenery },
```

Comment:
i think you should also add to the .d.ts file.

```
queryAction(action: "scenerysetrestricted", args: ScenerySetRestricted, callback?: (result: GameActionResult) => void): void;
executeAction(action: "scenerysetrestricted", args: ScenerySetRestricted, callback?: (result: GameActionResult) => void): void;

interface ScenerySetRestricted extends GameActionArgs {
        objectType: ObjectType;  // see ObjectType
        object: number;
        isRestricted: boolean;
    }
```
@Basssiiie , @Sadret could you confirm that that would be all that is needed.

Reply:
That should be all for the plugin d.ts yes, though one one minor suggestion to change
`interface ScenerySetRestricted` 
to
`interface ScenerySetRestrictedArgs` 
for consistency. 🙂